### PR TITLE
Add prerequisite module UI test

### PR DESCRIPTION
**Pull request by armstrongnate**
[run-nightly]

refs: MBL-13692
affects: student
release note: none

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: CanvasCore/CanvasCore/Modules/ModuleItem/ModuleItem+Network.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ extension ModuleItem {
     public static func moduleItemSequence(_ session: Session, courseID: String, moduleItemID: String) throws -> SignalProducer<JSONObject, NSError> {
         let path = "api/v1/courses/\(courseID)/module_item_sequence"
         let parameters = ["asset_id": moduleItemID, "asset_type": "ModuleItem"]
-        let request = try session.GET(path, parameters: parameters)
+        let request = try session.GET(path, parameters: parameters, paginated: false)
```
Comment (cobbal) [inline CanvasCore/CanvasCore/Modules/ModuleItem/ModuleItem+Network.swift]: Could this cause any performance issues for large courses?

File: Core/Core/Models/ModuleItemType.swift [hunk 2]
```diff
@@ -17,6 +17,7 @@
 //
 
 import Foundation
+import UIKit
```
Comment (cobbal) [inline Core/Core/Models/ModuleItemType.swift]: Is this supposed to be there?

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-13692](https://instructure.atlassian.net/browse/MBL-13692)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.41% | 93.41% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1017577&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: CanvasCore/CanvasCore/Modules/ModuleItem/ModuleItem+Network.swift (hunk 1 above)
Comment (armstrongnate) [reply CanvasCore/CanvasCore/Modules/ModuleItem/ModuleItem+Network.swift]: This particular api call isn't a paginated list at all. `session.GET` is dumb and will always append `per_page=99` unless we tell it not to.

File: Core/Core/Models/ModuleItemType.swift (hunk 2 above)
Comment (armstrongnate) [reply Core/Core/Models/ModuleItemType.swift]: Yeah I got a flaky build failure in here related to this file referencing `UIImage`. I figured this wouldn't hurt. Shouldn't technically be necessary I don't think.

File: Student/StudentUITests/Modules/ModulesTests.swift [hunk 3]
```diff
@@ -120,3 +121,73 @@ class ModulesTests: CoreUITestCase {
...
+        mockData(PostMarkModuleItemRead(courseID: "1", moduleID: "1", moduleItemID: "1"))
+        show("courses/1/modules")
+        XCTAssertEqual(Modules.module(index: 0).label(), "Module 1")
+        XCTAssertEqual(Modules.module(index: 1).label(), "Module 2. Status: Locked")
+        Modules.module(index: 1).tap()
+        app.find(labelContaining: "PREREQUISITE MODULES").waitToExist()
+        XCTAssertEqual(app.find(id: "module_cell_0_0").label(), "Module 1")
+        XCTAssertEqual(ModulesDetail.moduleItem(index: 0).label(), "Page 2. Type: Page. Status: Locked")
+        ModulesDetail.module(index: 0).tap()
+        ModulesDetail.moduleItem(index: 0).tap()
```
Comment (cobbal) [inline Student/StudentUITests/Modules/ModulesTests.swift]: test doesn't execute this line correctly for me. Race condition with loading maybe?

Comment (armstrongnate) [reply Student/StudentUITests/Modules/ModulesTests.swift]: I'll check. Maybe I broke something with a refactor or a merge.

Comment (armstrongnate) [reply Student/StudentUITests/Modules/ModulesTests.swift]: @cobbal the problem was a deadlock on the main thread. Should be good now.

**Pull request merged**